packages/grafana-ui/src/components/Table/TableNG/Cells/PillCell.tsx

```diff
@@ -0,0 +1,153 @@
+import { css } from '@emotion/css';
+import { Property } from 'csstype';
+import { useMemo } from 'react';
+
+import { GrafanaTheme2, isDataFrame, classicColors, colorManipulator } from '@grafana/data';
+import { TablePillCellOptions } from '@grafana/schema';
+
+import { useStyles2 } from '../../../../themes/ThemeContext';
+import { TableCellRendererProps } from '../types';
+
+export function PillCell({ value, field, justifyContent, cellOptions }: TableCellRendererProps) {
+  const styles = useStyles2(getStyles, justifyContent);
+
+  const pills = useMemo(() => {
+    if (!value) {
+      return [];
+    }
+
+    // Handle DataFrame - not supported for pills
+    if (isDataFrame(value)) {
+      return [];
+    }
+
+    // Handle different value types
+    const stringValue = String(value);
+
+    // Try to parse as JSON first
+    try {
+      const parsed = JSON.parse(stringValue);
+      if (Array.isArray(parsed)) {
+        // JSON array of strings
+        return parsed
+          .filter((item) => item != null && item !== '')
+          .map(String)
+          .map((text) => text.trim())
+          .filter((item) => item !== '');
+      }
+    } catch {
+      // Not valid JSON, continue with other parsing
+    }
+
+    // Handle CSV string
+    if (stringValue.includes(',')) {
+      return stringValue
+        .split(',')
+        .map((text) => text.trim())
+        .filter((item) => item !== '');
+    }
+
+    // Single value - strip quotes
+    return [stringValue.replace(/["'`]/g, '').trim()];
+  }, [value]);
+
+  if (pills.length === 0) {
+    return <div className={styles.cell}>-</div>;
+  }
+
+  return (
+    <div className={styles.cell}>
+      <div className={styles.pillsContainer}>
+        {pills.map((pill, index) => {
+          const bgColor = getPillColor(pill, cellOptions);
+          const textColor = colorManipulator.getContrastRatio('#FFFFFF', bgColor) >= 4.5 ? '#FFFFFF' : '#000000';
+
+          return (
+            <span
+              key={`${pill}-${index}`}
+              className={styles.pill}
+              style={{
+                backgroundColor: bgColor,
+                color: textColor,
+              }}
+            >
+              {pill}
+            </span>
+          );
+        })}
+      </div>
+    </div>
+  );
+}
+
+function isPillCellOptions(cellOptions: TableCellRendererProps['cellOptions']): cellOptions is TablePillCellOptions {
+  return cellOptions?.type === 'pill';
+}
+
+function getPillColor(pill: string, cellOptions: TableCellRendererProps['cellOptions']): string {
+  if (!isPillCellOptions(cellOptions)) {
+    return getDeterministicColor(pill);
+  }
+
+  const colorMode = cellOptions.colorMode || 'auto';
+
+  // Fixed color mode (highest priority)
+  if (colorMode === 'fixed' && cellOptions.color) {
```

Comment:
how useful is it to have all pills in the column be the same color?

we should support value mappings here.

Reply:
yeah, fixed was an easy one to add, all the components where there so it was an easy win, will it be used, we could track it but its not that much weight really. 

I'm happy to try to add in kv mapping, maybe in next iteration? I did have it at first, but it made it a bit more complicated. I didn't try with kv pairs tho, just a mapping similar to our other kv mapping we have.

What do you think @leeoniya , do we need kv pairs to move forward or is this a good first iteration?

Reply:
Added in the latest commits, have a look @leeoniya 😄 , added video to the main PR description.